### DISCUSSION TITLE
ESP32 with ENC28J60

### DISCUSSION THREAD

**Discussion by cleverpal-ai** [arendst/Tasmota#16130] [Q&A] [answered]
Can anybody advise how to make ESP32 (LuaNode32) friendly with Ethernet module ENC28J60 and Tasmota?
I need to understand the connection diagram and what should be configured in Tasmota to make ESP32 work with LAN. In my scenario Wi-Fi cannot be used. Thanks!

**Comment by sfromis** [comment]
Not sure about exact details, but it looks like ENC28J60 is not supported, as it is not mentioned in 
`#define ETH_TYPE          0                    // [EthType] 0 = ETH_PHY_LAN8720, 1 = ETH_PHY_TLK110/ETH_PHY_IP101, 2 = ETH_PHY_RTL8201, 3 = ETH_PHY_DP83848, 4 = ETH_PHY_DM9051, 5 = ETH_PHY_KSZ8081`
Of course, if it is compatible with one of the above....

**Comment by barbudor** [comment, marked as answer]
@SuperMaximus1984
This is definitively not supported
EN28J60 is completed Ethernet adapter circuit dedicated to be used with MCU That do not have an embedded Ethernet controller (such like with old Arduino ATmega etc ...)
The ESP32 already include a Ethernet controller to which you need to connect a PHY interface like the one listed above

Don't loose your time by trying to do that by yourself and get an ESP32 board with Ethernet interface already on-board such as:
- Olimex ESP32-EVB, ESP32-GATEWAY or ESP32-POE/ESP32-POE-ISO
- Wireless Tag ESP32-ETH01
